doc/12_onstack_policy.qbk

```diff
@@ -34,22 +37,23 @@ to
 or to
 
  // Stack-based implementation allocation
- struct Book : boost::impl_ptr<Book, impl_ptr_policy::onstack, char[64]> { ... };
+ struct Book : boost::impl_ptr<Book, impl_ptr_policy::onstack, void*[8], void*> { ... };
 
 with minimal or no disruption to the existing code. 
 
 Then, a restricted custom ['stack-based-implementation] policy (suggested by Giel van Schijndel) might be a further extension of impl_ptr_policy::onstack:
 
- template<typename impl_type, typename size_type>
- struct onstack_no_null : impl_ptr_policy::onstack<impl_type, size_type>
+ template<typename impl_type, typename size_type, typename align_type = void>
```

Comment:
Volodya, I was suggesting instead:

```
template<typename impl_type, typename storage_type>
```

i.e. Instead of independent `size_type` and `align_type` parameters that you are just going to use with `aligned_storage`, why not simply give the user the option to specify the storage type? They could supply `aligned_storage_t<X, Y>` as that type or something else.

Reply:
Hey, Glen, nice hearing from you. How is it in the crazy land of free?.. It's no nice and quiet down under. ;-) Thank you for keeping an eye on this. Truly appreciated. Yes, indeed, you are certainly right. What is currently happening with the 'onstack' policy is that I've been struggling with it lacking the expertise and real deployment cases to make sure it's legit. So, Giel kindly agreed to take over this policy. He has it deployed and, so, feels better, knows more how to proceed. So, I'll leave it for him to decide. From my (admittedly) layman view the need to specify already aligned storage seems scary. I very much want to simply indicate the size (like int[64]) and then take the default alignment. But again I lack real expertise and experience in the area.

Reply:
Giel, thank you for the submission. Much appreciated. Could you please have a read of Glen's suggestion. On one hand it kills two birds with one stone. On the other hand it does ask more from a layman person (like me) as then I'll need to know about alignment issues instead of relying on default. You decide. 

I have not merged your branch (nice and compact, thanks) yet because 

1) it fails to compile on my 64-bit -- static_assert in impl_onstack.cpp fails. Sizes reported are sizeof(OnStack)=48 and the other is 24. So, I erred on the safe side before merging.

2) You adjusted the docs. That is great. I really mean it. What I feel somewhat uncomfortable with is that in the docs you get too serious too soon. :-) For an uninitiated reader or a layman (like me) it feels scary. Say, I want to switch from allocator-based to stack-based. I.e. from 
boost::impl_ptr<Book, policy::copied, my_allocator>
to
boost::impl_ptr<Book, policy::onstack, ...>

Then, asking to specify the size (like char[64], int[16]) seems reasonable. familiar and not off-putting (again, a layman speaking here). You instead, "grab the bull by the horns" and list void*[8]... which a few will stumble upon... More so, you add the alignment as well. I suggest in the docs we take it slowly. :-) That is, we say "Hey, deploy onstack as easy as changing

boost::impl_ptr<Book, policy::copied, my_allocator>
to
boost::impl_ptr<Book, policy::onstack, int[16]>. Done."

Then, later in the chapter, as "true salesmen" we say "wait, there is more"... and you introduce more configurability -- alignment and all. That way a layman (like me) gets the basic knowledge from the previous paragraphs but then says "Gosh, it's getting scary" and skips the alignment part. An expert (like you) keeps reading... All benefit. What do you think?